src/index.ts

```diff
@@ -1 +1,535 @@
+            )?.snapshotName;
+            if (!snapshotName) throw new Error('Cannot reset device: No snapshot name specified.');
+
+            return timeout(platform.resetDevice(snapshotName), {
+                milliseconds: 5 * 60 * 1000,
+            }).catch(async (err) => {
+                if (!(err instanceof TimeoutError)) throw err;
+
+                // Sometimes, the Android emulator gets stuck and doesn't accept any commands anymore. In this case, we
+                // restart it.
+                await timeout(this.ensureDevice({ killExisting: true }), { milliseconds: 60 * 1000 });
+                await timeout(platform.resetDevice(snapshotName), { milliseconds: 5 * 60 * 1000 });
+            });
+        },
+        startAppAnalysis: async (appPath, options) => {
+            const appPathMain = getAppPathMain(appPath);
+            const appMeta = await parseAppMeta(appPathMain);
+            if (!appMeta) throw new Error(`Could not start analysis with invalid app: "${appPathMain}"`);
+
+            const res: AppAnalysisResult = {
+                app: appMeta,
+                traffic: {},
+            };
+
+            const installApp = () => platform.installApp(getAppPathAll(appPath).join(' '));
+            const uninstallApp = () => platform.uninstallApp(appMeta.id);
+
+            let inProgressTrafficCollectionName: string | undefined;
+            let mitmproxyState: { proc: ExecaChildProcess; harOutputPath: string; wireguardConf?: string } | undefined;
+
+            const cleanUpAppAnalysis = async () => {
+                killProcess(mitmproxyState?.proc);
+                platform._internal?.objectionProcesses?.forEach((proc) => killProcess(proc));
+            };
+
+            if (options?.resetApp) {
+                await uninstallApp();
+                await installApp();
+            }
+
+            if (options?.noSigint !== true) {
+                process.removeAllListeners('SIGINT');
+                process.on('SIGINT', async () => {
+                    await cleanUpAppAnalysis();
+                    process.exit();
+                });
+            }
+
+            return {
+                app: appMeta,
+
+                installApp,
+                setAppPermissions: (permissions) => platform.setAppPermissions(appMeta.id, permissions),
+                uninstallApp,
+                startApp: () => platform.startApp(appMeta.id),
+
+                startTrafficCollection: async (name) => {
+                    if (inProgressTrafficCollectionName)
+                        throw new Error(
+                            `Cannot start new traffic collection. Previous one "${inProgressTrafficCollectionName}" is still running.`
+                        );
+                    if (analysisOptions.platform === 'ios') throw new Error('Unimplemented'); // TODO: We don't have wireguard support on iOS, yet.
```

Comment:
True, but in this case I like to comment because it explains why we throw an error on iOS, What do you say, we only ever use TODO comments if we also create an issue referenced in it?

Reply:
I'm also very much in favour of making the error message longer than a single word anyway. :D